docs/tutorials/configure/pet-photographer.md

```diff
@@ -0,0 +1,730 @@
+    """
+    Registry.register_resource_creator(
+        Camera.SUBTYPE,
+        color_filter.ColorFilterCam.MODEL,
+        ResourceCreatorRegistration(
+            color_filter.ColorFilterCam.new_cam,
+            color_filter.ColorFilterCam.validate_config
+        )
+    )
+    module = Module.from_args()
+    module.add_model_from_registry(Camera.SUBTYPE, color_filter.ColorFilterCam.MODEL)
+    await module.start()
+
+if __name__ == "__main__":
+    asyncio.run(main())
+```
+
+{{% /tab %}}
+{{% tab name="Go"%}}
+<file>main.go</file> initializes and starts the colorfilter model of the camera component.
+
+  <details>
+    <summary>Click to view sample code from <file>main.go</file></summary>
+
+```go {class="line-numbers linkable-line-numbers"}
+// Package main is a module which serves the colorfilter custom module.
+package main
+
+import (
+  "context"
+
+  "github.com/edaniels/golog"
+  "go.viam.com/utils"
+
+  "github.com/viam-labs/modular-filter-examples/colorfilter"
+  "go.viam.com/rdk/components/camera"
+  "go.viam.com/rdk/module"
+)
+
+func main() {
+  utils.ContextualMain(mainWithArgs, module.NewLoggerFromArgs("colorfilter_module"))
+}
+
+func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) (err error) {
+  myMod, err := module.NewModuleFromArgs(ctx, logger)
+  if err != nil {
+    return err
+  }
+
+  err = myMod.AddModelFromRegistry(ctx, camera.API, colorfilter.Model)
+  if err != nil {
+    return err
+  }
+
+  err = myMod.Start(ctx)
+  defer myMod.Close(ctx)
+  if err != nil {
+    return err
+  }
+  <-ctx.Done()
+  return nil
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+For more information on creating your own module, read [Code your own modules to create modular resources](https://docs.viam.com/modular-resources/create/).
+
+## Code your filter resource model
+
+The code for this colorfilter camera model (<file>[color_filter.go](https://github.com/viam-labs/modular-filter-examples/blob/main/colorfilter/color_filter.go)</file>) or (<file>[color_filter.py](https://github.com/viam-labs/modular-filter-examples/blob/main/pycolorfilter/color_filter.py)</file>) is sourced from the full modular filter examples available on the [Viam GitHub](https://github.com/viam-labs/modular-filter-examples/tree/main).
+
+### Include subtype's required methods
+
+In order to code a new filter resource model, you must implement the required methods outlined in the <file>[client.go](https://github.com/viamrobotics/rdk/components/camera/client.go)</file> file in the corresponding resource's directory.
+In this case, the `rdk/components/camera` directory.
+
+Provide this as a file inside of your `colorfilter` module directory to serve as your module's client interface, <file>color_filter.go</file> or <file>color_filter.py</file>. Make sure you name your model with all lowercase letters for optimal performance with Viam's SDKs.
```

Comment:
```suggestion
Inside your model's folder `colorfiler`, create a file called <file>color_filter.go</file> or <file>color_filter.py</file> inside of your `colorfilter` directory and implement the required methods in it.
```

However I don't know why that is here. At this point they should already have created that folder because that's where the entry point file surely went? or not? Moving that further up.